### PR TITLE
docs: Mention privacy implications of reprovider settings 

### PR DESCRIPTION
**Pull request by ghost**
As stated by: https://docs.ipfs.tech/concepts/privacy-and-encryption/

Some reprovider options have better privacy than others, as this wasn't mentioned here, I wanted to mention it here.

### CODE REVIEW

**Review by Jorropo — COMMENTED** [review]
I dont think its good, this could give a false sens of security.
I think its best if you mention that all cids provided are considered public information. Without going into best and worst.

Better is relative and can still mean bad. However a non négligeable number of people think better imply some base level of the thing, which we dont claim is a thing without using tor or i2p.

Ill leave open for other maintainers to say what they think.

File: docs/config.md [hunk 1]
```diff
@@ -1287,9 +1287,15 @@ Type: `array[peering]`
...
+*Some options have better privacy than others.*
+
 - `"all"` - announce all CIDs of stored blocks
 - `"pinned"` - only announce pinned CIDs recursively (both roots and child blocks)
-- `"roots"` - only announce the root block of explicitly pinned CIDs
+- `"roots"` - only announce the root block of explicitly pinned CIDs (offers the best privacy)
+
+**DISCLAIMER: Assume that your CIDs are public information regardless of the option you choose.**
+
+Learn more on [IPFS Docs](https://docs.ipfs.tech/concepts/privacy-and-encryption/)
```
Comment (mishmosh) [inline docs/config.md:1298]:
```suggestion
Learn more about privacy and encryption on [IPFS Docs](https://docs.ipfs.tech/concepts/privacy-and-encryption/)
```

File: docs/config.md [hunk 2]
```diff
@@ -1254,9 +1254,15 @@ Type: `duration`
 
 Tells reprovider what should be announced. Valid strategies are:
 
+*Some options have better privacy than others.*
+
```
Comment (Jorropo) [inline docs/config.md:1258]:
```suggestion
```

File: docs/config.md [hunk 3]
```diff
@@ -1254,9 +1254,15 @@ Type: `duration`
...
 - `"all"` - announce all CIDs of stored blocks
 - `"pinned"` - only announce pinned CIDs recursively (both roots and child blocks)
-- `"roots"` - only announce the root block of explicitly pinned CIDs
+- `"roots"` - only announce the root block of explicitly pinned CIDs (offers the best privacy)
```
Comment (Jorropo) [inline docs/config.md:1261]:
```suggestion
- `"roots"` - only announce the root block of explicitly pinned CIDs
```

File: docs/config.md [hunk 4]
```diff
@@ -1254,9 +1254,15 @@ Type: `duration`
...
+
+**DISCLAIMER: Assume that your CIDs are public information regardless of the option you choose.**
```
Comment (Jorropo) [inline docs/config.md:1263]:
```suggestion
Kubo assumes that CIDs are public information. For example they are publicly announced in the network DHT.
```

**Review by Jorropo — CHANGES_REQUESTED** [review]
I don't like saying something and then adding a disclaimer.

Adding the link to the docs is nice thanks. :slightly_smiling_face:

**Pull request closed without merging**